habitat-lab/habitat/articulated_agent_controllers/humanoid_rearrange_controller.py

```diff
@@ -100,24 +100,27 @@ def __init__(
             self.walk_motion.displacement[-1] / self.walk_motion.num_poses
         )
 
-        # State variables
-        self.obj_transform = mn.Matrix4()
+        # These two matrices store the global transformation of the base
+        # as well as the transformation caused by the walking gait
+        # We initialize them to identity
+        self.obj_transform_offset = mn.Matrix4()
+        self.obj_transform_base = mn.Matrix4()
+
         self.prev_orientation = None
         self.walk_mocap_frame = 0
 
     def reset(self, position) -> None:
         """Reset the joints on the human. (Put in rest state)"""
-        self.obj_transform.translation = position + self.base_offset
+        self.obj_transform_offset = mn.Matrix4()
+        self.obj_transform_base.translation = position + self.base_offset
 
     def get_stop_pose(self):
         """
         Returns a stop, standing pose
         """
         joint_pose = self.stop_pose.joints
-        obj_transform = (
-            self.obj_transform
-        )  # the object transform does not change
-        return joint_pose, obj_transform
+        # the object transform does not change
+        return joint_pose, self.obj_transform_offset, self.obj_transform_base
```

Comment:
It seems returning `joint_pose, self.obj_transform_offset, self.obj_transform_base` is now common. Should we consider creating a NamedTuple or a class that holds these 3 variables?

Reply:
Resolved